### ISSUE TITLE
Lint uses of `format!` strings that are only used for later formatting

### ISSUE THREAD

**Issue by clarcharr** [Manishearth/rust-clippy#1717]
Essentially, we should warn on patterns like:

```
let msg = format!(...);
// ...
println!("{} ...", msg, ...);
```

Because this does an unnecessary allocation. In cases like this, we should suggest changing the `format!` to `format_args!` or refactoring the code so that it's not needed in the first place.

I found an example of this in [`arch-audit`](https://github.com/ilpianista/arch-audit/blob/be79f6984fb50f63de8c3de183d17a0431aa66ce/src/main.rs#L350) when I was working on the code a bit and I feel like it's definitely something that clippy could warn for. I'm not sure how much effort we really should put into this (e.g. somehow refactoring the `format!` into a `print!` or `write!` somewhere in the code) but at least warning for using the `format!` itself is a nice start.

**Comment by camsteffen** [comment]
We now have `format_in_format_args` but it could be enhanced to work with variables.
